src/components/Pages/index.js

```diff
@@ -26,31 +24,36 @@ class VmDetailPage extends React.Component {
     this.requestSent = false
   }
 
-  componentWillMount () {
-    if (Selectors.isFilterChecked()) {
-      this.props.getVms({ vmId: this.props.match.params.id })
+  fetchTheVmIfNeeded () {
+    const { match, vms, getVms } = this.props
+    let requested = false
+
+    if (!this.requestSent && !vms.getIn(['vms', match.params.id])) {
```

Comment:
Originally, the VM was forced to reload when the detail page is opened.

The idea is: immediately show what's in store and update once/if changes arrive. I would like to keep it.

Reply:
Agree, because store after initial VMs fetch doesn't have all data needed for detail page.